addons/cookoff/XEH_postInit.sqf

```diff
@@ -58,3 +58,50 @@
         [FUNC(detonateAmmunition), [_vehicle, _mags, _total], _delay] call CBA_fnc_waitAndExecute;
     };
 }, nil, ["Man","StaticWeapon"]] call CBA_fnc_addClassEventHandler;
+
+if (hasInterface) then {
+    GVAR(soundObjPool) = [];
+    [QFUNC(poolGetObject), {
+        private _pooledObject = GVAR(soundObjPool) param [0, [-999, objNull]];
+        _pooledObject params ["_time", "_obj"];
+        if (isNull _obj || _time >= time) then {
+            _obj = "#particlesource" createVehicleLocal [0, 0, 0];
+        } else {
+            GVAR(soundObjPool) deleteAt 0;
+        };
+        GVAR(soundObjPool) pushBack [time + 8, _obj];
+        _obj; // return
+    }] call CBA_fnc_compileFinal;
+
+    [QGVAR(playCookoffSound), {
+        params ["_obj", "_sound", "_maxDist"];
+        if (isNull _obj) exitWith {};
+        private _distance = _obj distance (positionCameraToWorld [0,0,0]);
+        if (_distance > _maxDist) exitWith {};
+
+        // Delay sound after Rule of SOS.
+        [{
+            params ["_obj", "_sound", "_maxDist"];
+            private _distance = _obj distance (positionCameraToWorld [0, 0, 0]); // update Value so that distance is correct to the current camera Position
+            private _sDistance = _distance + ((random 5) - 2.5);
+            private _disStep = [[["far", 1.5], ["mid", 1.3]] select (_sDistance < 952), ["close", 1]] select (_sDistance < 235);
+            _sound = format [QGVAR(%1_%2_%3), _sound, _disStep select 0, ceil (random 3)];
+            hintSilent _sound;
+            // fix for Vanilla Say3d Falloff
+            private _value = 1 - (_distance / _maxDist);
+            private _fakeValue = linearConversion ([[0, 0.7, _value, 1, 0.8], [0.7, 1, _value, 0.8, 0]] select (_value >= 0.7));
+            private _soundSource = call FUNC(poolGetObject);
+            _soundSource setPos (getPos _obj);
+            _soundSource say3D [_sound, (_distance / _fakeValue) * (_disStep select 1), 0.9 + (random 0.2)];
```

Comment:
Why not use [playSound3D](https://community.bistudio.com/wiki/playSound3D)? In my testing (in single player) it obeys the speed of sound, as in it takes time for the sound to propagate. According to the wiki page it also takes the Doppler effect into account, but I did not verify that.

Reply:
Doppler Effect != Speed of Sound and https://www.youtube.com/watch?v=fhsKUGzLHog
and https://github.com/acemod/ACE3/issues/9547 does prove that playSound3D does not use Speed Of Sound

Reply:
> Doppler Effect != Speed of Sound
I know that, I just thought it's something nice to have if possible.
I'll test `playSound3D` in MP on my end to see what it gives.

Reply:
Well.. both the bug report and my findings are correct.

`playSound3D` does obey speed of sound on the machine where `playSound3D` is called from. For others it's immediate.

However, if the following is executed, it plays fine:

```sqf
[["z\ace\addons\cookoff\sounds\heavy_crack_close.wss", objNull, false, getPosASL player, 5, 1, 1000, 0, true]] remoteExecCall ["playSound3D", 0]
```

We can make an event that calls it locally, would be the easiest solution imo.

Reply:
Yeah, that is definitely new, as this PR is over 6 years old. playSound3D didn't have SoS, nor did it have a local tag.

Additionally, we still can't blindly play the sound effects but still need to do a local measuring of the distance and choose the correct sound to play.